### PR TITLE
[varLib/featureVars] Insert rvrn lookups at beginning

### PR DESCRIPTION
**Pull request by behdad**
Fixes https://github.com/fonttools/fonttools/issues/2140
Fixes https://github.com/googlefonts/fontmake/issues/950

### CODE REVIEW

**Comment by behdad** [comment]
The part for renumbering of (Chain)Context lookups is untested.

File: Lib/fontTools/varLib/featureVars.py [hunk 1]
```diff
@@ -414,19 +415,42 @@ def makeSubstitutionsHashable(conditionalSubstitutions):
         condSubst.append((conditionSet, substitutions))
     return condSubst, sorted(allSubstitutions)
 
+class ShifterVisitor(TTVisitor):
```
Comment (anthrotype) [inline Lib/fontTools/varLib/featureVars.py:418]: Ha, nice use of a visitor!

File: Lib/fontTools/varLib/featureVars.py [hunk 2]
```diff
@@ -414,19 +415,42 @@ def makeSubstitutionsHashable(conditionalSubstitutions):
...
+    def __init__(self, shift):
+        self.shift = shift
+
+@ShifterVisitor.register_attrs(
+    (
+        (ot.SubstLookupRecord, ("LookupListIndex")),  # GSUB
+        (ot.PosLookupRecord, ("LookupListIndex")),  # GPOS
```
Comment (anthrotype) [inline Lib/fontTools/varLib/featureVars.py:425]: why not add `FeatureRecord` as well here? it has the same "LookupListIndex" attribute

Comment (behdad) [reply Lib/fontTools/varLib/featureVars.py:425]: `FeatureRecord.LookupListIndex` is a list of numbers, not just a number.

Comment (behdad) [reply Lib/fontTools/varLib/featureVars.py:425]: I mean, sure, I can add another block. Let me do that.

**Comment by behdad** [comment]
PTAL!

**Review by anthrotype — APPROVED** [review]
LGTM!

**Pull request merged**